src/dda/dda.c

```diff
@@ -0,0 +1,87 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   dda.c                                              :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ycarro <ycarro@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/22 14:48:17 by ycarro            #+#    #+#             */
+/*   Updated: 2023/02/23 17:42:22 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "dda.h"
+
+void	dda(char **map, t_player *p, t_ray	*ray_data, t_point *collision)
+{
+	(void)map;
+	(void)collision;
+	set_incrementor(ray_data, ray_data->to_cast.radians);
+	get_ray_data(ray_data, ray_data->to_cast.radians, p);
+	while (!it_collides(ray_data, map, collision))
+	{
+		if (ray_differs_in_x(ray_data))
+		{
+			ray_data->x_pos.x += (double)(ray_data->incrementor.x * 100);
```

Comment:
nice :)
i think it would be more readable if this was a function -> ray_increment_x()
or somthing like that

Reply:
Donee